deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/recurrent/LSTMHelpers.java

```diff
@@ -249,6 +251,15 @@ static public FwdPassReturn activateHelper(final Layer layer, final NeuralNetCon
                 currHiddenUnitActivations.muliColumnVector(timeStepMaskColumn);
                 currentMemoryCellState.muliColumnVector(timeStepMaskColumn);
             }
+           // boolean layerNormalization = true;
+            if (useLayerNormalization) {
+                Number stdev = currHiddenUnitActivations.stdNumber();
+                currHiddenUnitActivations = currHiddenUnitActivations.sub(Nd4j.mean(currHiddenUnitActivations));
+                if (stdev.floatValue() == stdev.floatValue()) {
+                    // don't divide by NaN. i.e., all zero activations.
+                    currHiddenUnitActivations = currHiddenUnitActivations.div(stdev);
+                }
```

Comment:
The test is testing for equality, not small differences. A float is equal to itself iff it is not a NaN (in Java, see 6 in http://stackoverflow.com/questions/1456566/how-do-you-test-to-see-if-a-double-is-equal-to-nan). 

I will refactor the floatValue() call to a variable to reduce the redundant computation.